go/src/koding/kites/kloud/provider/builder.go

```diff
@@ -311,13 +311,33 @@ func (bs *BaseStack) applyAsync(ctx context.Context, req *kloud.ApplyRequest) er
 		return err
 	}
 
+	// At this point resources have been built successfully on the terraform side.
+	// If updating stack or provisioning klient on remote hosts fails,
+	// we need to cleanup the stack; if debug: true was set in the stack template
+	// for any of the instance resources, the stack won't be destroyed
+	// so it's possible to troubleshoot it post-mortem.
+	if !bs.Debug {
+		defer func() {
+			if err != nil {
+				_, e := tfKite.Destroy(&tf.TerraformRequest{
```

Comment:
> Just checked the code again and after this commit we broke that support.

This commit did not break it, as it did not work in the first place - the goroutine was executed before / during terraform process, so when state file was not even created (it failed with `state.tf: no such file or directory` error).

Right now `Try building again` button always creates stack from scratch. Making it rebuild instead (do an incremental apply) should be safe with one caveat: if building stack fails on `checking connectivity` then rebuild would be a dummy operation.

I agree and I lean towardds reverting this and fixing the terraformer to keep the state file on failures.